core/src/main/java/com/linecorp/armeria/common/logging/DefaultTextLogFormatter.java

```diff
@@ -0,0 +1,218 @@
+        requireNonNull(sanitizers, "sanitizers,");
+
+        final Set<RequestLogProperty> availableProperties = log.availableProperties();
+        if (!availableProperties.contains(RequestLogProperty.REQUEST_START_TIME)) {
+            return "{}";
+        }
+
+        final String requestCauseString;
+        if (availableProperties.contains(RequestLogProperty.REQUEST_CAUSE) && log.requestCause() != null) {
+            requestCauseString = String.valueOf(log.requestCause());
+        } else {
+            requestCauseString = null;
+        }
+
+        final String sanitizedHeaders;
+        if (availableProperties.contains(RequestLogProperty.REQUEST_HEADERS) && log.requestHeaders() != null) {
+            sanitizedHeaders = sanitize(log, sanitizers.headersSanitizer(), log.requestHeaders());
+        } else {
+            sanitizedHeaders = null;
+        }
+
+        final String sanitizedContent;
+        if (availableProperties.contains(RequestLogProperty.REQUEST_CONTENT) && log.requestContent() != null) {
+            sanitizedContent = sanitize(log, sanitizers.contentSanitizer(), log.requestContent());
+        } else {
+            sanitizedContent = null;
+        }
+
+        final String sanitizedTrailers;
+        if (availableProperties.contains(RequestLogProperty.REQUEST_TRAILERS) &&
+            !log.requestTrailers().isEmpty()) {
+            sanitizedTrailers = sanitize(log, sanitizers.trailersSanitizer(), log.requestTrailers());
+        } else {
+            sanitizedTrailers = null;
+        }
+
+        try (TemporaryThreadLocals tempThreadLocals = TemporaryThreadLocals.acquire()) {
+            final StringBuilder buf = tempThreadLocals.stringBuilder();
+            buf.append("{startTime=");
+            TextFormatter.appendEpochMicros(buf, log.requestStartTimeMicros());
+
+            if (availableProperties.contains(RequestLogProperty.REQUEST_LENGTH)) {
+                buf.append(", length=");
+                TextFormatter.appendSize(buf, log.requestLength());
+            }
+
+            if (availableProperties.contains(RequestLogProperty.REQUEST_END_TIME)) {
+                buf.append(", duration=");
+                TextFormatter.appendElapsed(buf, log.requestDurationNanos());
+            }
+
+            if (requestCauseString != null) {
+                buf.append(", cause=").append(requestCauseString);
+            }
+
+            buf.append(", scheme=");
+            if (availableProperties.contains(RequestLogProperty.SCHEME) && log.scheme() != null) {
+                buf.append(log.scheme().uriText());
+            } else if (availableProperties.contains(RequestLogProperty.SESSION)) {
+                buf.append(SerializationFormat.UNKNOWN.uriText())
+                   .append('+')
+                   .append(log.sessionProtocol() != null ? log.sessionProtocol().uriText() : "unknown");
+            }
+
+            if (availableProperties.contains(RequestLogProperty.NAME) && log.name() != null) {
```

Comment:
ditto for the json formatter

```suggestion
            if (availableProperties.contains(RequestLogProperty.NAME)) {
```